classes/class-wc-connect-nux.php

```diff
@@ -83,5 +94,168 @@ public function register_add_service_to_zone_pointer( $pointers ) {
 			);
 			return $pointers;
 		}
+
+		public function get_jetpack_install_status() {
+			$status = Jetpack_Install_Status::UNINSTALLED;
+
+			if( 0 === validate_plugin( 'jetpack/jetpack.php' ) ) {
+				$status = Jetpack_Install_Status::INSTALLED;
+			}
+
+			if ( ! class_exists( 'Jetpack_Data' ) ) {
+				// installed, but not activated
+				return $status;
+			}
+
+			if ( defined( 'JETPACK_DEV_DEBUG' ) ) {
+				$status = Jetpack_Install_Status::DEV;
+				return $status;
+			}
+
+			if ( ! method_exists( 'Jetpack_Data', 'get_access_token' )
+				|| ! class_exists( 'Jetpack' )
+			) {
+				$status = Jetpack_Install_Status::ERROR;
+				return $status;
+			}
+			$status = Jetpack_Install_Status::ACTIVATED;
+
+
+			$user_token = Jetpack_Data::get_access_token( JETPACK_MASTER_USER );
+			if ( $user_token
+				&& is_object( $user_token )
+				&& isset( $user_token->external_user_id )
+			) {
+				$status = Jetpack_Install_Status::CONNECTED;
+			};
+
+			return $status;
+		}
+
+		public function show_relevant_nux_notices() {
+			$jetpack_install_status = $this->get_jetpack_install_status();
+			switch ( $jetpack_install_status ) {
+				case Jetpack_Install_Status::ACTIVATED:
+					add_action( 'admin_notices', array( $this, 'show_banner_before_connection_welcome' ) );
+					add_action( 'admin_notices', array( $this, 'show_banner_before_connection_get_access' ) );
+					break;
+				case Jetpack_Install_Status::CONNECTED:
+					add_action(
+						'admin_notices',
+						array( $this, 'show_banner_after_connection_see_how_it_works' )
+					);
+					break;
+			}
+		}
+
+		public function show_banner_before_connection_get_access( $content ) {
+			if( function_exists( 'get_current_screen' ) ) {
+				$screen = get_current_screen();
+			}
+			if ( ! isset( $screen ) ) {
+				return;
+			}
+			if( ! (
+				'edit' === $screen->parent_base
+				&& 'post' === $screen->base
+				&& 'product' === $screen->post_type
+			) ) {
+				return;
+			}
+			if (  'add' === $screen->action ) {
+				$redirect = admin_url( 'post-new.php?post_type=product' );
+			} else {
+				global $post;
+				$redirect = get_edit_post_link( $post->ID, false );
```

Comment:
You can save accessing the global `$post` by passing in `null` instead.

See:
https://developer.wordpress.org/reference/functions/get_edit_post_link/
https://developer.wordpress.org/reference/functions/get_post/